langchain/document_loaders/onedrive.py

```diff
@@ -0,0 +1,219 @@
+"""Loader that loads data from OneDrive"""
+
+import logging
+import os
+import tempfile
+from enum import Enum
+from pathlib import Path
+from typing import Dict, List, Optional, Type, Union
+
+try:
+    from O365 import Account, FileSystemTokenBackend
+    from O365.drive import Drive, Folder
+except ImportError:
+    raise ValueError(
+        "o365 package not found, please install it with `pip install o365`"
+    )
+
+from pydantic import BaseModel, BaseSettings, Field, FilePath, SecretStr
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+from langchain.document_loaders.onedrive_file import OneDriveFileLoader
+
+SCOPES = ["offline_access", "Files.Read.All"]
+logger = logging.getLogger(__name__)
+
+
+class _OneDriveSettings(BaseSettings):
+    client_id: str = Field(..., env="O365_CLIENT_ID")
+    client_secret: SecretStr = Field(..., env="O365_CLIENT_SECRET")
+
+    class Config:
+        env_prefix = ""
+        case_sentive = False
+        env_file = ".env"
+
+
+class _OneDriveTokenStorage(BaseSettings):
+    token_path: FilePath = Field(Path.home() / ".credentials" / "o365_token.txt")
+
+
+class _FileType(str, Enum):
+    DOC = "doc"
+    DOCX = "docx"
+    PDF = "pdf"
+
+
+class _SupportedFileTypes(BaseModel):
+    file_types: List[_FileType]
+
+    def fetch_mime_types(self) -> Dict[str, str]:
+        mime_types_mapping = {}
+        for file_type in self.file_types:
+            if file_type.value == "doc":
+                mime_types_mapping[file_type.value] = "application/msword"
+            elif file_type.value == "docx":
+                mime_types_mapping[
+                    file_type.value
+                ] = "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
+            elif file_type.value == "pdf":
+                mime_types_mapping[file_type.value] = "application/pdf"
+        return mime_types_mapping
+
+
+class OneDriveLoader(BaseLoader, BaseModel):
```

Comment:
lets expose this (and the file loader) in document_loaders/__init__.py

Reply:
done.